lib/elixir/test/elixir/kernel/errors_test.exs

```diff
@@ -361,10 +361,6 @@ defmodule Kernel.ErrorsTest do
     assert_compile_fail CompileError,
       "nofile:1: unknown key :age for struct Kernel.ErrorsTest.GoodStruct",
       '%#{GoodStruct}{age: 27} = %{}'
-
-    assert_compile_fail CompileError,
-      "nofile:1: unknown key ^field for struct Kernel.ErrorsTest.GoodStruct",
-      '%#{GoodStruct}{^field => 27} = %{}'
```

Comment:
@josevalim with the changes in `elixir_exp`, `%GoodStruct{^field => 27}` doesn't fail with "unknown key" but with "unbound variable ^field". We can make this test

```elixir
'field = :foo; %#{GoodStruct}{^field => 27} = %{}'
```

and assert on the "unknown key" error message if you prefer. At that point I think this file is a reasonable place though, otherwise did you have any suggestions on where to put it?